### PR TITLE
refactor!: simplify discv5 initialization

### PR DESCRIPTION
**Pull request by wemeetagain**
**Motivation**

Simplification of initialization of discv5 for the purpose of simplifying passing the enr across a thread boundary

### CODE REVIEW

File: packages/beacon-node/src/network/options.ts [hunk 1]
```diff
@@ -20,22 +19,14 @@ export interface NetworkOptions
...
-
 export const defaultNetworkOptions: NetworkOptions = {
   maxPeers: 55, // Allow some room above targetPeers for new inbound peers
   targetPeers: 50,
   discv5FirstQueryDelayMs: 1000,
   localMultiaddrs: ["/ip4/0.0.0.0/tcp/9000"],
   bootMultiaddrs: [],
   mdns: false,
-  discv5: defaultDiscv5Options,
+  discv5: null,
```
Comment (dapplion) [inline packages/beacon-node/src/network/options.ts:30]: Note that defaulting to null means discv5 is disabled, is that the intention? If so we should add a comment like "discv5 = null disables discv5 by default"

File: packages/beacon-node/src/network/peers/peerManager.ts [hunk 2]
```diff
@@ -61,6 +61,12 @@ const ALLOWED_NEGATIVE_GOSSIPSUB_FACTOR = 0.1;
 // The Node should compute a recomended value every interval and log a warning
 // to terminal if it deviates significantly from the user's settings
 
+export type LodestarDiscv5Opts = Omit<IDiscv5CreateOptions, "enr" | "multiaddr"> & {
```
Comment (dapplion) [inline packages/beacon-node/src/network/peers/peerManager.ts]: We should not extend `IDiscv5CreateOptions` with Omit, Either select the few properties we want to expose or just use a new object. From looking into it we just need

```ts
export type Discv5Opts = {
  enr: string;
  bindAddr: string;
  bootEnrs: string[];
};
```

**Comment by wemeetagain** [comment]
@dapplion ready for re-review

**Review by dapplion — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.9.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.9.0) :tada:

**Pull request merged**